pkg/scripts/csv-to-couchdb.ts

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env -S deno run --allow-read --allow-net
+
+import Denomander from "https://deno.land/x/denomander@0.9.1/mod.ts";
+import { readCSV } from "https://deno.land/x/csv/mod.ts";
+import ProgressBar from "https://deno.land/x/progress@v1.2.4/mod.ts";
+import { BufReader } from "https://deno.land/std/io/buffer.ts";
+import { bulkDocs, createDatabase } from "../couchdb/couchdb.ts"
+
+/**
+  * Counts lines in a csv file for progress bar calculations
+  * @param {string} filePath - csv file path 
+  * @return {Number} - Number of lines in file
+  */
+
+const howManyLines = async function (filePath: string) {
+    let lineCount: number = 0;
+    var file;
+    console.log({ filePath })
+    try {
+        file = await Deno.open(filePath);
+        const bufReader = new BufReader(file);
+        while (await bufReader.readLine()) {
+            lineCount++
+        };
+    } finally {
+        if (file) file.close();
+    }
+    return lineCount;
+}
+
+//#region progress bar
+/**
+  * Renders progress bar
+  * @param {ProgressBar} progress - progress bar instance 
+  * @param {Number} progress - chunk size for incrementation
+  */
+let completed = 0;
+function downloading(progress: ProgressBar, chunkSize: number) {
+    if (completed <= progress.total!) {
+        progress.render(completed += chunkSize);
+    }
+}
+//#endregion progress bar
+
+
+//#region csv to couch
+/**
+  * Parses CSV file and saves docs into couchdb
+  * @param {string} filePath - csv file path 
+  * @param {string} columns - headers of doc columns 
+  * @param {string} couchdbName - name of couchdb database (default: books)
+  * @param {string} couchdbURL - url of couchdb database (default: http://admin:admin@localhost:5984)
+  * @param {string} columnSeparator - separator used for columns (default: ,)
+  * @param {ProgressBar} progress - progress bar instance
+  * @return {Iterator} - An iterator over the chunks
+  */
+
+async function CSVToCouch({ filePath, columns, couchdbName = "books",
+    couchdbURL = "http://admin:admin@localhost:5984", columnSeparator = ",", chunkSize = 1000, progress }: {
+        filePath: string,
+        columns: string, couchdbName: string,
+        couchdbURL: string, columnSeparator: string,
+        chunkSize: number,
+        progress: ProgressBar
+
+    }) {
+
+    // create couch client with endpoint
+    const columnsArr = columns.split(',')
+
+    // csv file options
+    const options = { columnSeparator: columnSeparator }
+
+
+    try {
+        await createDatabase(couchdbName, couchdbURL)
+        const f = await Deno.open(filePath);
+        console.log("Saving docs bulk...")
+
+        for await (const chunk of inChunks(readCSV(f, options), chunkSize, columnsArr)) {
+            await bulkDocs(chunk, couchdbName, couchdbURL)
+            downloading(progress, chunkSize);
```

Comment:
We could initialize and update a `completed` variable in this funciton, and pass it to ~`downloading`~ `printProgress`.

Reply:
We could initialize `completed` here but it has to be updated inside the `printProgress` function, right? @silviot

Reply:
I don't think `printProgress` should update a counter: it should only print info about the counter it receives. Updating the counter should happen here IMHO.